feed-rs/src/xml/mod.rs

```diff
@@ -522,11 +533,13 @@ impl XmlEvent {
 
     // Creates a new event corresponding to an XML text node
     fn text<R: BufRead>(text: &BytesText, reader: &Reader<R>) -> XmlResult<Option<XmlEvent>> {
-        let text = text.unescape_and_decode(reader)?;
+        let escaped_text = reader.decoder().decode(text)?;
```

Comment:
In theory this could be shortened to [`text.unescape`](https://docs.rs/quick-xml/0.25.0/quick_xml/events/struct.BytesText.html#method.unescape) but for reasons I was unable to determine, that results in a decoding error in the `parser::rss0::tests::test_0_91_encoding_2` test.